src/components/LoginButton.jsx

```diff
@@ -1,37 +1,59 @@
 import React, { useEffect, useState } from 'react';
 import { Link } from 'react-router-dom';
 import firebase from 'firebase';
+import { CircularProgress, makeStyles } from '@material-ui/core';
 import Button from '@material-ui/core/Button';
 import ProfileMenu from './ProfileDropdown';
 
+const useStyles = makeStyles(() => ({
+  progress: {
+    display: 'block',
+    margin: 'auto',
+  },
+}));
+
 export default function LoginButton() {
   const [isSignedIn, setIsSignedIn] = useState(!!firebase.auth().currentUser);
+  const [loading, setLoading] = useState(true);
+  const [error] = useState();
+  const classes = useStyles();
+
+  function updateUserStatus(user) {
+    setLoading(false);
+    setIsSignedIn(!!user);
+  }
 
   useEffect(
-    () => firebase.auth().onAuthStateChanged((user) => setIsSignedIn(!!user)),
+    () => firebase.auth().onAuthStateChanged((user) => updateUserStatus(user)),
     []
   );
 
   const signUserOut = () => firebase.auth().signOut();
   const user = firebase.auth().currentUser;
 
-  return isSignedIn ? (
-    <ProfileMenu
-      signOut={signUserOut}
-      name={user.displayName}
-      email={user.email}
-      photo={user.photoURL}
-    />
-  ) : (
-    <Button
-      variant="contained"
-      color="primary"
-      component={Link}
-      to={{
-        state: { showLoginDialog: true },
-      }}
-      replace>
-      Login
-    </Button>
+  return (
+    <>
+      {error?.toString() ||
+        (loading && <CircularProgress className={classes.progress} />) ||
+        (isSignedIn ? (
```

Comment:
Btw, this logic probably makes more sense outside of LoginButton. When I think of what LoginButton is supposed to be, I think it's a button for logging in and maybe logging out. But in reality this component is more than just the button so I suggest moving this logic to HeaderNavMenu and getting rid of the `LoginButton` component as it's no longer just a button.

Reply:
Wait so the entire LoginButton functionality now be contained within the HeaderNavMenu and no longer have a LoginButton component?

Reply:
I removed it but the login button component is used in another component the ScholarshipsMadeSimple component which I did git rm to the LoginButton and now the code broke and the component is not in my trash bin. Can you just undo the latest commit and go back to the previous, thanks

Reply:
It comes down to naming and purpose. `LoginButton` as a name suggests it's a button for logging in. But the logic of "should I show a ProfileMenu or a LoginButton?" does not really belong in a component named `LoginButton`. That logic belongs in `HeaderNavMenu` when it's deciding which child element to show. 

Regarding the issue with ScholarshipsMadeSimple, I think you just need to replace `LoginButton` with the same `Button` you now have in `HeaderNavMenu`.

Reply:
Ok yeah I see exactly what you're referring to, that makes sense. Yeah I replaced it with the button that is in HeaderNavMenu and that did more. Now again as you mentioned above the logic for if a button is displayed or the avatar will happen in the `ScholarshipsMadeSimple`.

Reply:
Alright the logic for showing a ProfileMenu or LoginButton on the `ScholarshipsMadeSimple` component is working accordingly.